Error while checking build/O2/fullCI for b5b80bf76a7e1064da19e93dc766e7f175a1d502 at 2023-07-04 18:28:
```
## sw/BUILD/GEANT4_VMC-latest/log
CMake Error at /sw/slc8_x86-64/CMake/v3.23.1-13/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/11608/b5b80bf76a7e1064da19e93dc766e7f175a1d502/build_O2_fullCI/pretty.html).

--- a/Framework/Core/include/Framework/ExternalFairMQDeviceProxy.h
+++ b/Framework/Core/include/Framework/ExternalFairMQDeviceProxy.h
@@ -104,7 +104,8 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* label,
                                                    std::vector<OutputSpec> const& outputs,
                                                    const char* defaultChannelConfig,
                                                    InjectorFunction converter,
-                                                   uint64_t minSHM = 0);
+                                                   uint64_t minSHM = 0,
+                                                   bool sendTFcounter = false);
 
 DataProcessorSpec specifyFairMQDeviceOutputProxy(char const* label,
                                                  Inputs const& inputSpecs,

--- a/Framework/Core/src/ExternalFairMQDeviceProxy.cxx
+++ b/Framework/Core/src/ExternalFairMQDeviceProxy.cxx
@@ -425,7 +425,8 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* name,
                                                    std::vector<OutputSpec> const& outputs,
                                                    char const* defaultChannelConfig,
                                                    InjectorFunction converter,
-                                                   uint64_t minSHM)
+                                                   uint64_t minSHM,
+                                                   bool sendTFcounter)
 {
   DataProcessorSpec spec;
   spec.name = strdup(name);
@@ -437,7 +438,7 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* name,
   // The Init method will register a new "Out of band" channel and
   // attach an OnData to it which is responsible for converting incoming
   // messages into DPL messages.
-  spec.algorithm = AlgorithmSpec{[converter, minSHM, deviceName = spec.name](InitContext& ctx) {
+  spec.algorithm = AlgorithmSpec{[converter, minSHM, deviceName = spec.name, sendTFcounter](InitContext& ctx) {
     auto* device = ctx.services().get<RawDeviceService>().device();
     // make a copy of the output routes and pass to the lambda by move
     auto outputRoutes = ctx.services().get<RawDeviceService>().spec().outputs;
@@ -593,7 +594,7 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* name,
       }
     };
 
-    auto runHandler = [dataHandler, minSHM](ProcessingContext& ctx) {
+    auto runHandler = [dataHandler, minSHM, sendTFcounter](ProcessingContext& ctx) {
       static RateLimiter limiter;
       auto device = ctx.services().get<RawDeviceService>().device();
       limiter.check(ctx, std::stoi(device->fConfig->GetValue<std::string>("timeframes-rate-limit")), minSHM);
@@ -624,7 +625,7 @@ DataProcessorSpec specifyExternalFairMQDeviceProxy(char const* name,
               timingInfo.creation = dph->creation;
             }
             dataHandler(timingInfo, parts, 0, ci);
-            if (ctx.services().active<o2::monitoring::Monitoring>()) {
+            if (sendTFcounter) {
               ctx.services().get<o2::monitoring::Monitoring>().send(o2::monitoring::Metric{(uint64_t)timingInfo.tfCounter, "df-sent"}.addTag(o2::monitoring::tags::Key::Subsystem, o2::monitoring::tags::Value::DPL));
             }
           }

--- a/run/o2sim_mctracks_proxy.cxx
+++ b/run/o2sim_mctracks_proxy.cxx
@@ -146,6 +146,14 @@ InjectorFunction o2simKinematicsConverter(std::vector<OutputSpec> const& specs, 
     }
 
     if (*totalEventCounter == static_cast<size_t>(nevents)) {
+      if (nPerTF > 0) {
+        // send accumulated messages if the limit is reached
+        ++(*TFcounter);
+        sendOnChannel(device, *MCHeadersMessageCache.get(), channelRetriever(specs[0], *TFcounter), *TFcounter);
+        sendOnChannel(device, *MCTracksMessageCache.get(), channelRetriever(specs[1], *TFcounter), *TFcounter);
+        MCHeadersMessageCache->Clear();
+        MCTracksMessageCache->Clear();
+      }
       // I am done (I don't expect more events to convert); so tell the proxy device to shut-down
       stop = true;
     }

--- a/run/o2sim_mctracks_proxy.cxx
+++ b/run/o2sim_mctracks_proxy.cxx
@@ -202,7 +202,7 @@ WorkflowSpec defineDataProcessing(ConfigContext const& configcontext)
 
   auto proxy = specifyExternalFairMQDeviceProxy("o2sim-mctrack-proxy",
                                                 outputs,
-                                                channelspec.c_str(), f);
+                                                channelspec.c_str(), f, 0, true);
   // add monitoring service to be able to report number of timeframes sent for the rate limiting to work
   proxy.requiredServices.push_back(o2::framework::ArrowSupport::arrowBackendSpec());
   // if aggregation is requested, set the enumeration repetitions to aggregation size

